transmogrify/dexterity/tests/testconverters.py

```diff
@@ -1,12 +1,68 @@
 # -*- coding: utf-8 -*-
+from ftw.builder import Builder
+from ftw.builder import create
 from plone.app.textfield import RichText
+from plone.formwidget.contenttree import ObjPathSourceBinder
 from transmogrify.dexterity import converters
 from transmogrify.dexterity.interfaces import IDeserializer
+from transmogrify.dexterity.testing import TRANSMOGRIFY_DEXTERITY_FUNCTIONAL_TESTING
+from z3c.relationfield.relation import RelationValue
+from z3c.relationfield.schema import RelationChoice
+from z3c.relationfield.schema import RelationList
 import pprint
 import unittest
 import zope.testing
 
 
+class TestRelationDeserializer(unittest.TestCase):
+
+    layer = TRANSMOGRIFY_DEXTERITY_FUNCTIONAL_TESTING
+
+    relation = RelationChoice(
+        title=u"Relation",
+        source=ObjPathSourceBinder(),
+        required=False,
+    )
+
+    relation_list = RelationList(
+        title=u"Relation List",
+        default=[],
+        value_type=RelationChoice(title=u"Relation",
+                                  source=ObjPathSourceBinder()),
+        required=False,
+    )
+
+    def test_deserialize_relation(self):
+        folder = create(Builder('folder'))
+        deserializer = IDeserializer(self.relation)
+        value = deserializer(folder, None, None)
+
+        self.assertTrue(isinstance(value, RelationValue))
+        self.assertEqual(folder, value.to_object)
+
+    def test_deserialize_none_relation(self):
+        deserializer = IDeserializer(self.relation)
+        value = deserializer(None, None, None)
+
+        self.assertIsNone(value)
+
+    def test_deserialize_relation_list(self):
+        folder = create(Builder('folder'))
+        deserializer = IDeserializer(self.relation_list)
+        value = deserializer([folder], None, None)
+
+        self.assertEqual(1, len(value))
```

Comment:
Hm i think this would be incorrect. `RelationList` ist only used in the schema definition, i.e. `RelationList` of `RelationChoice`. The actual value is a `list` of `RelationValue` objects.